cmd/frontend/internal/search/search.go

```diff
@@ -374,6 +396,21 @@ func fromStrPtr(s *string) string {
 	return *s
 }
 
+// withDecoration hydrates event match with decorated hunks for a corresponding file match.
+func withDecoration(ctx context.Context, eventMatch streamhttp.EventMatch, internalResult result.Match, kind string, contextLines int) streamhttp.EventMatch {
+	if _, ok := internalResult.(*result.FileMatch); !ok {
+		return eventMatch
+	}
+	if _, ok := eventMatch.(*streamhttp.EventContentMatch); !ok {
+		return eventMatch
+	}
+	if kind == "html" {
+		eventMatch.(*streamhttp.EventContentMatch).Hunks = DecorateFileHunksHTML(ctx, internalResult.(*result.FileMatch))
+	}
```

Comment:
We can avoid the extra typecast

```suggestion
	ecm, ok := eventMatch.(*streamhttp.EventContentMatch); 
	if !ok {
		return eventMatch
	}
	if kind == "html" {
		ecm.Hunks = DecorateFileHunksHTML(ctx, internalResult.(*result.FileMatch))
	}
```